chainer/serializers/hdf5.py

```diff
@@ -47,8 +47,13 @@ def __call__(self, key, value):
         ret = value
         if isinstance(value, cuda.ndarray):
             value = cuda.to_cpu(value)
-        arr = numpy.asarray(value)
-        compression = None if arr.size <= 1 else self.compression
+        if value is None:
+            # use Empty to represent None
+            arr = h5py.Empty('f')
```

Comment:
I used `Empty` to represent `None` because h5py does not support serializing `None`. `Empty` can have a dtype unlike `None`, which is not used and can be arbitrary. There is no reason of using `'f'` here; it is random any other values are also acceptable. The deserialization code does not rely on the choice.